doc/hooks/patch-activation-hooks.section.md

```diff
@@ -0,0 +1,51 @@
+
+# `patchActivationBash`, `patchActivationCsh`, `patchActivationFish`, `patchActivationPosix` {#sec-patchActivationHooks}
+
+These hooks provide utilities to patch
+shell scripts meant to be sourced (e.g. the "activate" scripts).
+
+It inserts two sections of codes into the beginning and the end of the script:
+the former inserts a `PATH` prefix, while the latter clears it up.
+This way, the original commands will have access to the executables needed,
+but how `PATH` variable is changed by the script remains the same after patching.
+
+A stack is used to record the inserted `PATH` prefixes when patched multiple times,
+so that a patched activation script can correctly source another patched script.
+
+The utilities are called as
+
+```sh
+patchActivation<Shellname> <file> <PATH-prefix>
+```
+
+Example usage:
+
+```nix
+{ lib, stdenv, patchActivationBash}:
+stdenv.mkDerivation {
+
+  # ...
+
+  nativeBuildInputs = [
+    patchActivationBash
+  ];
+
+  postFixup = ''
+    patchActivationBash $out/bin/activate-foo.zsh ${lib.makeBinPath [ coreutils man which ]}
+  '';
+}
+```
+
+## Compatible shell languages {#sec-patchActivationHooks-shells}
+
+As the patch is language-specific, different hooks are plovided for different shells:
+
+ - `patchActivationBash`: Bash, Ksh, Zsh and other shells supporting the Bash-like parameter expansions.
+ - `patchActivationCsh`: C Shell (Csh) scripts.
+ - `patchActivationFish`: Friendly Interactive Shell (Fish) scripts.
+ - `patchActivationPosix`: POSIX-conformant shells supporting the limited parameter expansions specified by the POSIX standard.
```

Comment:
This should move to the top, as in my suggestion.

It's not mandatory, but it would be worthwhile to link to the manuals or front pages of each specific shell.

Reply:
> This should move to the top, as in the other comment.

Addressed.

> It's not mandatory, but it would be worthwhile to link to the manuals or front pages of each specific shell.

Sounds great! There's currently no section for shell scripts. Should we create one?

Reply:
> Should we create one?

Of course, feel free to open a PR.